### PR TITLE
Fix rtc bug and add tests

### PR DESCRIPTION
**Pull request by bradfrizzell**

### CODE REVIEW

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ export class RealTimeConfigManager {
     }
 
     const isGloballyValid = this.isValidCalloutForConsentState(this.rtcConfig_);
-    this.rtcConfig_.urls = this.rtcConfig_.urls.filter(
+    this.rtcConfig_.urls = (this.rtcConfig_.urls || []).filter(
```
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: urls is undefined/null, this means you'll be allocating it.  Just want to make sure that is your intent as opposed to just wrapping in an if

File: extensions/amp-a4a/0.1/test/test-real-time-config-manager.js [hunk 2]
```diff
@@ -635,6 +635,30 @@ describes.realWin('real-time-config-manager', {amp: true}, env => {
       expect(rtc.rtcConfig_).to.deep.equal(expectedRtcConfig);
     });
 
+    it('should handle empty urls array', () => {
+      rtc.consentState_ = CONSENT_POLICY_STATE.UNKNOWN;
+      rtc.rtcConfig_.urls = [];
+      expect(rtc.modifyRtcConfigForConsentStateSettings()).not.to.throw;
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/test/test-real-time-config-manager.js]: These tests are faulty, this method never throws.

Comment (jridgewell) [reply extensions/amp-a4a/0.1/test/test-real-time-config-manager.js]: Nevermind, this seem to have changed from the initial PR.

File: extensions/amp-a4a/0.1/real-time-config-manager.js (hunk 1 above)
Comment (bradfrizzell) [reply extensions/amp-a4a/0.1/real-time-config-manager.js]: yep

**Pull request merged**